core/include/traccc/edm/cluster.hpp

```diff
@@ -9,31 +9,46 @@
 
 // Project include(s).
 #include "traccc/edm/cell.hpp"
+#include "traccc/geometry/pixel_data.hpp"
 
 // System include(s).
 #include <functional>
 #include <vector>
 
 namespace traccc {
 
-using position_estimation = std::function<vector2(channel_id, channel_id)>;
-using signal_modeling = std::function<scalar(scalar)>;
+/// Function for signal modelling
+inline scalar signal_cell_modelling(scalar signal_in) {
+    return signal_in;
+}
+
+/// Function for pixel segmentation
+inline vector2 position_from_cell(cell c, pixel_data pixel, bool is_default) {
```

Comment:
You should just pass the "cluster ID" object to this function. No need to pass the `pixel_data` and `bool` parameters individually. The function should be allowed to use any identification characteristic of the detector module to figure out how to calculate X-Y coordinates for a cell.

Reply:
done